### ISSUE TITLE
Look into and implement Auth Service

### ISSUE THREAD

**Issue by a-stacey** [data61/Mirza#385]

**Comment by a-stacey** [comment]
Once the PR (#417) is merged the following will still be outstanding:
- [ ] Create a way for the tests to authenticate and reinclude tests.
- [ ] Dynamically fetch public key.
- [ ] Remove obsolete user fields from the user table / new user etc (and remove promoted user hack so that we don't need to assume there is a specific company in the DB for the auth process to work).
- [ ] Remove Scrypt as a dependency.
- [ ] Remove authDataToTokenTodoRemove.
- [ ] See what can be done about encodeJWT.
- [ ] Add a way of associating a user with a compan(y/ies).
- [ ] Remove SCS dependace on BR and hence auth hacks there.
- [ ] Add the audience claim (auth0 clientid) to the 2 empty "" locations used in tests.

(Some of the above are mutually overlapping, but listing for completeness simplicity to try to make sure everything is covered)

**Comment by a-stacey** [comment]
Note to self, changing the BR so that the company mapping happens in a separate table and adding an "associate user with company endpoint" is highest priority. Also change the create business endpoint so that it creates an association with the user that created the business (this is the inital user...and then removes the need for bootstrapping).

**Comment by paulsamways** [comment]
Look at #416

**Comment by a-stacey** [comment]
We are going to put this on hold for now, because the 2 issues outstanding are still things that should be fixed, but it doesn't exactly make sense to do this now because we don't have a workable solution for the citrus data at the moment and points to issues with or architecture.
